server/armeria-server/src/main/scala/sttp/tapir/server/armeria/ArmeriaServerRequest.scala

```diff
@@ -0,0 +1,65 @@
+package sttp.tapir.server.armeria
+
+import com.linecorp.armeria.common.HttpRequest
+import com.linecorp.armeria.server.ServiceRequestContext
+import java.net.InetSocketAddress
+import scala.collection.JavaConverters._
+import sttp.model.{Header, Method, QueryParams, Uri}
+import sttp.tapir.model.{ConnectionInfo, ServerRequest}
+import scala.collection.immutable.Seq
+
+private[armeria] final class ArmeriaServerRequest(ctx: ServiceRequestContext) extends ServerRequest {
+  private lazy val request: HttpRequest = ctx.request
+
+  lazy val connectionInfo: ConnectionInfo = {
+    val remotePort = ctx.remoteAddress[InetSocketAddress]().getPort
+    val clientAddress = InetSocketAddress.createUnresolved(ctx.clientAddress().getHostAddress, remotePort)
+    ConnectionInfo(
+      Some(ctx.localAddress[InetSocketAddress]()),
+      Some(clientAddress),
+      Some(ctx.sessionProtocol().isTls)
+    )
+  }
+
+  override lazy val method: Method = MethodMapping.fromArmeria(request.method())
+
+  override lazy val protocol: String = ctx.sessionProtocol().toString
+
+  override lazy val uri: Uri = Uri(ctx.request().uri())
+
+  override lazy val headers: Seq[Header] = HeaderMapping.fromArmeria(request.headers())
+
+  override def header(name: String): Option[String] = Option(request.headers().get(name))
+
+  override def underlying: Any = ctx
+
+  override val pathSegments: List[String] = {
+    // ctx.path() always starts with '/'.
+    if (ctx.path() == "/") {
+      Nil
+    } else {
+      // Armeria does not decode '%2F' in a path into '/' by default.
```

Comment:
hm I guess if the path contains `%2F`, this should be treated as a single path segment? Was this decoding needed for some test to pass?

Reply:
`d1/d2/r4.txt` was sent as `:path=/d1%2Fd2%2Fr4.txt`.
https://github.com/softwaremill/tapir/blob/937f983f6bec8bd10b44b21471e8879e28efacd6/server/tests/src/main/scala/sttp/tapir/server/tests/ServerStaticContentTests.scala#L460
Armeria does not decode `%2f` to `/` for interop with other libraries such gRPC-JSON transcoding.
https://github.com/googleapis/googleapis/blob/02710fa0ea5312d79d7fb986c9c9823fb41049a9/google/api/http.proto#L257-L258

Reply:
Done: https://github.com/softwaremill/tapir/pull/1868

Reply:
Thanks! Let me merge master and remove the workaround.